scripts/operations/system_recovery/rapid_rebuild.sh

```diff
@@ -0,0 +1,156 @@
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+set -o pipefail
+
+# Print an error message and exit
+function err_exit {
+  echo "ERROR: $*" >&2
+  exit 1
+}
+
+# Must run from m001
+if [[ $(hostname) != "ncn-m001" ]]; then
+  err_exit "Must be run on m001"
+fi
+
+# Remove any strings 'registry.local' or 'packages.local' from any lines in /etc/hosts
+# Add or replace the pit line to include 'registry.local' and 'packages.local'
+cp /etc/hosts /etc/hosts."$(date +%Y%m%d_%H%M%S)"
+IP=$(grep ncn-m001.nmn /etc/hosts | awk '{print $1}')
+pitregistry="${IP}      pit.nmn pit registry.local packages.local"
+sed -i -E 's/registry.local//g' /etc/hosts
+sed -i -E 's/packages.local//g' /etc/hosts
+grep -q pit /etc/hosts && sed -i "s/.*pit.*/${pitregistry}/" /etc/hosts || echo ${pitregistry} >> /etc/hosts
+
+# Exit if required services are not running -- dnsmasq will be checked later
+systemctl is-active --quiet nexus.service \
+  && systemctl is-active --quiet minio.service \
+  && systemctl is-active --quiet basecamp.service \
+  && systemctl is-active --quiet apache2.service \
+  && systemctl is-active --quiet conman.service \
+  && systemctl is-active --quiet apparmor.service \
+  && echo "Pit disk services are running" || err_exit "Not all pit disk services are running"
+
+# Check that needed environment variables are set
+env_vars=("IPMI_PASSWORD" "SYSTEM_NAME")
+for each in "${env_vars[@]}"; do
+  if [ -z "${!each}" ]; then
+    err_exit "$each is not defined"
+  fi
+done
+
+# Reset all the NCN keys
+truncate --size=0 /root/.ssh/known_hosts 2>&1
+grep -oP "(ncn-\w+)" /etc/hosts | sort -u | xargs -t -i ssh-keyscan -H \{\} >> /root/.ssh/known_hosts
+
+# Disable/stop etcd on m001 and delete the database
+systemctl disable --now etcd.service
+rm -rf /var/lib/etcd/*
+
+# Temp backup of files
+mkdir -p /tmp/backup_kubernetes
+mv /etc/kubernetes/* /tmp/backup_kubernetes/
+rm -rf /etc/kubernetes/*
```

Comment:
Is the `rm` needed?  Everything was moved out of the directory in the previous command, so there shouldn't be anything left.  If there is some other reason for the `rm`, it might be worth a quick comment.

Reply:
fixed